pkg/migration/controllers/migration.go

```diff
@@ -1499,10 +1513,120 @@ func (m *MigrationController) getParsedLabels(
 	return m.getParsedMap(labels, clusterPair)
 }
 
+// updateOwnerReferenceOnPVC updates owner reference on PVC objects
+// on destination cluster
+func (m *MigrationController) updateOwnerReferenceOnPVC(
+	migration *stork_api.Migration,
+	objects []runtime.Unstructured,
+	clusterPair *stork_api.ClusterPair,
+	pvcsWithOwnerRef []v1.PersistentVolumeClaim,
+) error {
+	remoteConfig, err := getClusterPairSchedulerConfig(migration.Spec.ClusterPair, migration.Namespace)
+	if err != nil {
+		return err
+	}
+	remoteAdminConfig := remoteConfig
+	// Use the admin cluter pair for cluster scoped resources if it has been configured
+	if migration.Spec.AdminClusterPair != "" {
+		remoteAdminConfig, err = getClusterPairSchedulerConfig(migration.Spec.AdminClusterPair, m.migrationAdminNamespace)
+		if err != nil {
+			return err
+		}
+	}
+	adminClient, err := kubernetes.NewForConfig(remoteAdminConfig)
+	if err != nil {
+		return err
+	}
+	remoteInterface, err := dynamic.NewForConfig(remoteConfig)
+	if err != nil {
+		return err
+	}
+	remoteAdminInterface := remoteInterface
+	if migration.Spec.AdminClusterPair != "" {
+		remoteAdminInterface, err = dynamic.NewForConfig(remoteAdminConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	ruleset := m.getDefaultRuleSet()
+	for _, srcPvc := range pvcsWithOwnerRef {
+		destPvc, err := adminClient.CoreV1().PersistentVolumeClaims(srcPvc.GetNamespace()).Get(context.TODO(), srcPvc.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		// Skip updating if destination PVC has owner reference enabled
+		if len(destPvc.GetOwnerReferences()) > 0 {
+			continue
+		}
+
+		destPvcOwnersList := make([]metav1.OwnerReference, 0)
+
+		// This loop finds the owner reference for the current pvc owner
+		for _, pvcOwner := range srcPvc.GetOwnerReferences() {
+			var dynamicClient dynamic.ResourceInterface
+			ownerName := ""
+			for _, o := range objects {
+				metadata, err := meta.Accessor(o)
+				if err != nil {
+					return err
+				}
+				objectType, err := meta.TypeAccessor(o)
+				if err != nil {
+					return err
+				}
+				// Found the source object which is the parent of the source pvc resource
+				if metadata.GetName() == pvcOwner.Name && objectType.GetKind() == pvcOwner.Kind {
+					ownerName = metadata.GetName()
+					resource := &metav1.APIResource{
+						Name:       ruleset.Pluralize(strings.ToLower(objectType.GetKind())),
+						Namespaced: len(metadata.GetNamespace()) > 0,
+					}
+					if resource.Namespaced {
+						dynamicClient = remoteInterface.Resource(
+							o.GetObjectKind().GroupVersionKind().GroupVersion().WithResource(resource.Name)).Namespace(metadata.GetNamespace())
+					} else {
+						dynamicClient = remoteAdminInterface.Resource(
+							o.GetObjectKind().GroupVersionKind().GroupVersion().WithResource(resource.Name))
+					}
+					break
+				}
+			}
+
+			if len(ownerName) == 0 {
+				return fmt.Errorf("could not find owner reference for pvc : %v", srcPvc.GetName())
```

Comment:
Instead of returning an error lets continue. This is to handle a case where some user explicitly disabled migrating CRs by stork and only wants to migrate PV, PVC objects. In such cases we can skip updating the ownership for now.

Reply:
Done